### ISSUE TITLE
[R] Missing predict() method for object of class lgb.CVBooster

### ISSUE THREAD

**Issue by minimaxir** [Microsoft/LightGBM#213]
The only `predict()` method exists for object of class `lgb.Booster`; for the output of a `lgb.cv()` training function, `predict()` throws an error:

```
Error in UseMethod("predict") : 
  no applicable method for 'predict' applied to an object of class "c('lgb.CVBooster', 'R6')"
```

**Comment by Laurae2** [comment]
Cross-validation does not produce a model but is only used to check for the metric evolution in a CV. You must train a model separately to use predict. If you want to access all the models separately, you just need to use a loop to train separately from each train fold and predict on its respective test fold.

Also, lgb.cv does not produce a lgb.Booster class. It produes a lgb.CVBooster class.

**Comment by minimaxir** [comment]
> You must train a model separately to use predict. If you want to access all the models separately, you just need to use a loop to train separately from each train fold and predict on its respective test fold.

That appears to be correct: the `lgb.CVBooster` object indeed contains a `booster` vector which has a list of `nfolds - 2` number of `lgb.Booster` objects.

The `xgboost` package has a method parameter to predict on the models resulting from a `xgb.cv`, however looking at the documentation closer for that package, the resulting prediction is for each of the folds, so it's less helpful.

Closing.

**Comment by naarkhoo** [comment]
theoretically, one should be able to predict - using the best model within the cross-validation (of course with the price of few data point).
